drivers/note/notectl_driver.c

```diff
@@ -0,0 +1,273 @@
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <errno.h>
+
+#include <nuttx/fs/fs.h>
+#include <nuttx/note/notectl_driver.h>
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int notectl_ioctl(struct file *filep, int cmd, unsigned long arg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct file_operations notectl_fops =
+{
+  NULL,          /* open */
+  NULL,          /* close */
+  NULL,          /* read */
+  NULL,          /* write */
+  NULL,          /* seek */
+  notectl_ioctl, /* ioctl */
+  NULL           /* poll */
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+  , 0            /* unlink */
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: notectl_ioctl
+ ****************************************************************************/
+
+static int notectl_ioctl(struct file *filep, int cmd, unsigned long arg)
+{
+  int ret = -ENOSYS;
+
+  /* Handle the ioctl commands */
+
+  switch (cmd)
+    {
+      /* NOTECTL_START
+       *      - Start note recoding
+       *        Argument: Ignored
+       */
+
+      case NOTECTL_START:
+        {
+          struct note_filter_mode_s mode;
+
+          sched_note_filter_mode(&mode, NULL);
+          mode.enable = true;
```

Comment:
We can do the same thing(NOTECTL_START/NOTECTL_STOP) with NOTECTL_SETMODE, so why not remove these two ioctl?